ofrak_core/ofrak/core/entropy/entropy.py

```diff
@@ -20,34 +17,11 @@
 
 C_LOG_TYPE = ctypes.CFUNCTYPE(None, ctypes.c_uint8)
 
-try:
-    _lib_entropy = ctypes.cdll.LoadLibrary(
-        os.path.abspath(os.path.join(os.path.dirname(__file__), "entropy.so.1"))
-    )
-    ENTROPY_FUNCTION = _lib_entropy.entropy
-
-    ENTROPY_FUNCTION.argtypes = (
-        ctypes.c_char_p,
-        ctypes.c_size_t,
-        ctypes.c_char_p,
-        ctypes.c_size_t,
-        C_LOG_TYPE,
-    )
-    ENTROPY_FUNCTION.restype = ctypes.c_int
-except OSError:
-    ENTROPY_FUNCTION = None  # type: ignore
-
-
-class _EntropyCTypesTool(ComponentExternalTool):
-    def __init__(self):
-        # TODO: Add docs page on building entropy.so.1
-        super().__init__("entropy.so.1", None, None, None)
 
-    def is_tool_installed(self) -> bool:
-        return ENTROPY_FUNCTION is not None
-
-
-_ENTROPY_SO_DEPENDENCY = _EntropyCTypesTool()
+try:
+    from .entropy_c import entropy_c as entropy_func
+except:
+    from ofrak.core.entropy.reference_entropy import entropy_func
```

Comment:
I put the logging at runtime when actually calculating entropy, to avoid giving the user superfluous warnings about a different implementation of functionality they aren't even using (then if they do use it, they get the warning).